static/app/views/dashboardsV2/widgetLibrary/widgetCard.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import styled from '@emotion/styled';
+
+import Button from 'app/components/button';
+import Card from 'app/components/card';
+import {IconAdd, IconCheckmark} from 'app/icons';
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+
+import {miniWidget} from '../utils';
+
+import {WidgetTemplate} from './data';
+
+type Props = {
+  widget: WidgetTemplate;
+  setSelectedWidgets: (widgets: WidgetTemplate[]) => void;
+  selectedWidgets: WidgetTemplate[];
+};
+
+function WidgetLibraryCard({selectedWidgets, widget, setSelectedWidgets}: Props) {
+  const selectButton = (
+    <StyledButton
+      type="button"
+      icon={<IconAdd size="small" isCircled color="gray300" />}
+      onClick={() => {
+        const updatedWidgets = selectedWidgets.slice().concat(widget);
+        setSelectedWidgets(updatedWidgets);
+      }}
+    >
+      {t('Select')}
+    </StyledButton>
+  );
+
+  const selectedButton = (
+    <StyledButton
+      type="button"
+      icon={<IconCheckmark size="small" isCircled color="gray300" />}
+      onClick={() => {
+        const removedWidgetIndex = selectedWidgets.indexOf(widget);
+        if (removedWidgetIndex !== -1) {
+          const clonedSelectedWidgets = [...selectedWidgets];
+          clonedSelectedWidgets.splice(removedWidgetIndex, 1);
+          setSelectedWidgets(clonedSelectedWidgets);
+        }
```

Comment:
nit: I think you might be able to simplify by using `filter` here:
```selectedWidgets.filter(selected => widget !== selected)```
`filter` also does not mutate the original array